### PR TITLE
Migrate pkg/kubelet/util to structured logging

### PR DESCRIPTION
**Pull request by navidshaikh**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate `pkg/kubelet/util` to structured logging

#### Which issue(s) this PR fixes:
Part of https://github.com/kubernetes/kubernetes/issues/98976 https://github.com/kubernetes/enhancements/issues/1602

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
- KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging

### CODE REVIEW

File: pkg/kubelet/util/manager/watch_based_manager.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 		// we limit ourselves to just quickly stop the reflector here.
 		if c.isImmutable(object) {
 			if item.stop() {
-				klog.V(4).Infof("Stopped watching for changes of %q/%q - object is immutable", namespace, name)
+				klog.V(4).InfoS("Stopped watching for changes - object is immutable", klog.KRef(namespace, name))
```
Comment (yuzhiquan) [inline pkg/kubelet/util/manager/watch_based_manager.go:211]:
```
klog.V(4).InfoS("Stopped watching for changes - object is immutable", "obj", klog.KRef(namespace, name))
```

**Review by adisky — COMMENTED** [review]
/lgtm

**Comment by navidshaikh** [comment]
This is ready to go. /cc @ehashman @serathius

File: pkg/kubelet/util/nodelease.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func SetNodeOwnerFunc(c clientset.Interface, nodeName string) func(lease *coordi
 					},
 				}
 			} else {
-				klog.Errorf("failed to get node %q when trying to set owner ref to the node lease: %v", nodeName, err)
+				klog.ErrorS(err, "Failed to get node when trying to set owner ref to the node lease", "node", nodeName)
```
Comment (serathius) [inline pkg/kubelet/util/nodelease.go:49]:
```suggestion
				klog.ErrorS(err, "Failed to get node when trying to set owner ref to the node lease", "nodeName", nodeName)
```

Comment (ehashman) [reply pkg/kubelet/util/nodelease.go:49]: I think node is fine?

Comment (navidshaikh) [reply pkg/kubelet/util/nodelease.go:49]: looks fine to me too, looking at `node` key [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#use-klogkobj-and-klogkref-for-kubernetes-objects)

excerpts:
For non-namespaced object we can pass empty string to namespace argument
```
func updateNode(nodeName string) {
   ...
   klog.InfoS("Updated node", "node", klog.KRef("", nodeName))
}
```

can change to `KRef` way but the result would be same, wdyt ?

Comment (ehashman) [reply pkg/kubelet/util/nodelease.go:49]: I think this is a fine MVP and we can leave this as is for now, this is in beta after all. We'll have time to tweak later.

Comment (serathius) [reply pkg/kubelet/util/nodelease.go:49]: Oh, your're right. Let's use `"node", klog.KRef("", nodeName)`. In text format there is no difference, but in Json output will differ. 
```suggestion
				klog.ErrorS(err, "Failed to get node when trying to set owner ref to the node lease", "node", klog.KRef("", nodeName))
```

Comment (navidshaikh) [reply pkg/kubelet/util/nodelease.go]: updated to use `klog.KRef("", nodeName))`

**Pull request merged**